pkg/apis/core/validation/validation.go

```diff
@@ -2816,6 +2816,45 @@ func validatePodResourceClaimSource(claimSource core.ClaimSource, fldPath *field
 	return allErrs
 }
 
+func validateLivenessProbe(probe *core.Probe, fldPath *field.Path) field.ErrorList {
```

Comment:
would be great to add e2e tests for sidecars with liveness, readiness, and startup probe. For liveness and readiness those can be added to all other readiness and liveness tests we already have.

Reply:
Maybe, we can just use `pull-kubernetes-e2e-gce-cos-alpha-features`.